server.go

```diff
@@ -1212,6 +1214,30 @@ func newServer(listenAddrs []net.Addr, chanDB *channeldb.DB,
 		}
 	}
 
+	if len(cfg.ExternalHosts) != 0 {
+		s.hostAnn = netann.NewHostAnnouncer(netann.HostAnnouncerConfig{
+			Hosts:           cfg.ExternalHosts,
+			RefreshInterval: time.Minute * 5,
+			LookupHost: func(host string) (net.Addr, error) {
+				return lncfg.ParseAddressString(
+					host, strconv.Itoa(defaultPeerPort),
```

Comment:
This should handle the node running on a different port.

Reply:
I don't think the port matters here, we only want to see if the _host name_ resolves to a new IP or not.